tools/autograd/gen_autograd.py

```diff
@@ -111,6 +111,12 @@ def load_aten_declarations(path):
                                               for arg in declaration['arguments']]
         declaration['type_method_args'] = [arg['name'] for arg in declaration['arguments']]
         declaration['api_name'] = declaration['name']
+        # NB: keep this in sync with common_with_cwrap.py
+        if declaration.get('overload_name'):
```

Comment:
see, if you had a common function to compute this, you wouldn't need to worry about keeping this in sync :).

Reply:
Can I even import any of the crap from ATen python scripts though? I feel like the only thing we share right now is CodeTemplate...

Reply:
that's a good question I don't know the answer to -- it feels like a good thing to fix if it doesn't work, though!

Reply:
Looking at the imports, it seems that only CodeTemplate is shared, and you have to go through some shenanigans to get to it:

```
try:
    from src.ATen.code_template import CodeTemplate
except ImportError:
    from tools.shared.module_loader import import_module
    CodeTemplate = import_module('code_template', 'aten/src/ATen/code_template.py').CodeTemplate

```

Reply:
@ezyang the shenanigans are in some places [but not others](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/gen.py#L19)... not sure why.

Reply:
files in ATen don't need these shenanigans as they are in the same network